### ISSUE TITLE
Body background image is not rendered

### ISSUE THREAD

**Issue by filipe-norte-red** [Metrological/WebKitForWayland#1249]
Under some circumstances, a background image set for the body element does not render on screen. The attached tests.zip contains the example page "test.image.html" that reproduces the issue as follows:

In a working case, screen background is set to blue (image) with a green rectangle on bottom right corner and a small video window in the top left corner:
![ok](https://github.com/WebPlatformForEmbedded/WPEWebKit/assets/89521141/eab5b1d6-25e0-45db-be43-b2846ea5c18f)

In the non-working case, the background remains black:
![nok](https://github.com/WebPlatformForEmbedded/WPEWebKit/assets/89521141/5713120f-13bd-4983-935b-88b91cb4904d)

The difference between the working case and non-working case is the z-index of the "videoDiv" div element that is set to 1 for the working case (see comments inside the html file for reference).

This issue is observed with wpe.2.38 and and 2.28. Observed on Amlogic and Realtek platforms.

Additional observations:
- Sometimes the page is correctly rendered after being loaded, but shortly after, the background goes black. This seems related with the video starting to play. When the z-index as mentioned above is set to 1, the page gets correctly rendered as soon as playback starts
- The example page "test.color.html" shows same issue as above when using a background color in the body element instead of an image, except that removing DOCTYPE line on the top of the file leads to correct rendering of the page (see comments inside the html file)

[tests.zip](https://github.com/WebPlatformForEmbedded/WPEWebKit/files/13513269/tests.zip)

**Comment by filipe-norte-red** [comment]
Additional note:
Currently, background color is being set as follows for the webview on startup:
```
WebKitColor transparent = { 0.0f, 0.0f, 0.0f, 0.0f };                                                                                                                                                                                                                                                               
webkit_web_view_set_background_color(m_view, &transparent); 
```
If this configuration / call is not done, test.color.html renders correctly

**Comment by modeveci** [comment]
@magomez , I know you are working on this. As further details, we need a solution for 2.28 as well, because it is a blocker for HBBTV certification. FYI

**Comment by magomez** [comment]
This is a bug that affects all ports upstream. I've opened https://bugs.webkit.org/show_bug.cgi?id=266478 to handle it.

**Comment by filipe-norte-red** [comment]
Thanks, @magomez , for the help with this issue (an others)! Is it possible to raise the priority on the upstream ticket as this affects hbbtv cert?
Thanks

CC: @emutavchi

**Comment by magomez** [comment]
Wow this is is tricky issue. Not only it reproduces in all the versions, but also the problem is not exactly the same in each of them, as the code has been changing a lot. This needs to be properly fixed upstream, but I think I've been able to create a patch to workaround the problem for 2.38. @filipe-norte-red can you give this patch a test? It fixes the test cases you attached on my side.

[1249.patch.txt](https://github.com/WebPlatformForEmbedded/WPEWebKit/files/13990846/1249.patch.txt)

**Comment by magomez** [comment]
The patch is for 2.38 BTW. If it works I'll check how to do something similar for 2.28.

**Comment by filipe-norte-red** [comment]
Thank you, @magomez for the investigation and patch! 
I tried it with the original use cases and it solved the issue. Did a bit more testing on some apps and running some other test cases and did not notice any obvious regressions with these changes.
Would it be viable to merge the patch to 2.38 branch from your end?

**Comment by magomez** [comment]
@filipe-norte-red I've just commited and pushed the patch to the wpe-2.38 and wpe-2.28 branches. I wasn't able to reproduce the issue on 2.28 on the rpi even with the patch, but I hope it will fix the problem on your side anyway.
I've also ported this to 2.42 and pushed it as well, so it doesn't bother us when upgrading to the new release.

Please verify that everything is ok and close this if that's the case. Thanks!

**Comment by magomez** [comment]
Do we have any feedback about this? Is the problem solved and the issue can be closed?

**Comment by filipe-norte-red** [comment]
Apologies for the late reply, @magomez . Yes, I confirm it is solved on 2.38. I was not able to confirm it on 2.28 yet. Will try to confirm it in the next days.

**Comment by modeveci** [comment]
@filipe-norte-red As it is solved and confirmed on 2.38, and no update for 2.28. I am closing this ticket for now. Feel free to re-open if it is needed. Regards
